core/src/poh_timing_reporter.rs

```diff
@@ -0,0 +1,157 @@
+use {
+    solana_metrics::{datapoint::PohTimingPoint, datapoint_info},
+    solana_sdk::clock::Slot,
+    std::collections::HashMap,
+};
+
+/// A PohTimestamp records timing of the events during the processing of a slot
+/// the validator
+#[derive(Debug, Clone, Copy)]
+pub struct SlotPohTimestamp {
+    /// Slot start time from poh
+    pub start_time: u64,
+    /// Slot end time from poh
+    pub end_time: u64,
+    /// Last shred received time from block producer
+    pub full_time: u64,
+}
+
+/// Default value (all zeros)
+impl Default for SlotPohTimestamp {
```

Comment:
Why not just `[derive(Default)]` instead of manually doing the same?

Reply:
Fixed.